test/e2e/suite_test.go

```diff
@@ -169,41 +169,132 @@ func cleanupTestNamespace(namespace string, kubeClient kubernetes.Interface) err
 	return nil
 }
 
-func createCluster(clusterName, kubeConfigPath, controlPlane, clusterContext string) error {
-	err := clusterProvider.Create(clusterName, cluster.CreateWithKubeconfigPath(kubeConfigPath))
+func createKwokCluster(clusterName string, kubeConfigPath string) (clusterContext string, err error) {
+	var nodeResource = `
+apiVersion: v1
+kind: Node
+metadata:
+  annotations:
+    node.alpha.kubernetes.io/ttl: "0"
+  labels:
+    beta.kubernetes.io/arch: ` + runtime.GOARCH + `
+    beta.kubernetes.io/os: linux
+    kubernetes.io/arch: ` + runtime.GOARCH + `
+    kubernetes.io/hostname: ` + clusterName + `-control-plane
+    kubernetes.io/os: linux
+    kubernetes.io/role: agent
+    node-role.kubernetes.io/agent: ""
+    type: kwok
+  name: ` + clusterName + `-control-plane
+status:
+  allocatable:
+    cpu: 32
+    memory: 256Gi
+    pods: 110
+  capacity:
+    cpu: 32
+    memory: 256Gi
+    pods: 110
+  nodeInfo:
+    architecture: ` + runtime.GOARCH + `
+    bootID: ""
+    containerRuntimeVersion: ""
+    kernelVersion: ""
+    kubeProxyVersion: fake
+    kubeletVersion: fake
+    machineID: ""
+    operatingSystem: linux
+    osImage: ""
+    systemUUID: ""
+  phase: Running
+`
+
+	kwokRuntime := "docker"
+	hostAddress := "host.docker.internal" // Require Docker 18.03
+	if runtime.GOOS == "linux" {
```

Comment:
Could you explain why use binary runtime only on linux?

Reply:
https://www.downloadkubernetes.com/

Kubernetes only provides binary files for Linux.